docs/general/adapters.md

```diff
@@ -194,6 +194,43 @@ The user could pass in `include=**`.
 
 We recommend filtering any user-supplied includes appropriately.
 
+#### Include Parameters
+
+You may wish to include related resources based on the `include` request
+parameter (See [fetching_includes](http://jsonapi.org/format/#fetching-includes). You can opt-in to this pattern:
+
+```ruby
+class PostSerializer < ActiveModel::Serializer
+  associations_via_include_param(true)
```

Comment:
Ok, thanks for understanding.  I'm going to block this implementation while we consider either a better one or an intermediate that let's you patch it yourself while we work on it in the gem

B mobile phone

> On Jun 10, 2016, at 8:41 AM, Lee Richmond <notifications@github.com> wrote:
> 
> In docs/general/adapters.md:
> 
> > @@ -194,6 +194,43 @@ The user could pass in `include=**`.
> >  
> >  We recommend filtering any user-supplied includes appropriately.
> >  
> > +#### Include Parameters
> > +
> > +You may wish to include related resources based on the `include` request
> > +parameter (See [fetching_includes](http://jsonapi.org/format/#fetching-includes). You can opt-in to this pattern:
> > +
> > +```ruby
> > +class PostSerializer < ActiveModel::Serializer
> > +  associations_via_include_param(true)
> So, the problem this solves is wanting to make a relationship link without loading the relationship, or to render the relationship id and type without loading the relationship?
> 
> Quick follow up note I hope clears up confusion - making the link without loading the relationship is kinda an independent issue (this is why @NullVoxPopuli and I arrived at making this code backwards-compatible, and just taking that performance hit in this scenario - so it can be addressed independently of this PR). This same problem exists for #1750 (I tried to sum up in #1750 (comment)). The way inline-relationships are loaded is an issue that impacts this PR, but it's not specific to or caused by this PR, if that makes sense.
> 
> I don't personally use inline associations so it's not actually applicable to me, was just trying to solve this use case at the same time when I ran into it. As you can see I am obsessed with unnecessary DB hits 😆
> 
> Avoiding the automatic rendering of /id/type conditionally based on the ?include parameter - and the performance hit that comes with that automatic rendering - is the goal of this PR.
> 
> Once again, I really do appreciate your time looking at this @bf4. As this gem is unusable to me without this, please let me know how I can help! Even if that is stepping back and letting you tackle this differently 😄
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>